serializers/src/glTF/2.0/Extensions/KHR_materials_clearcoat.ts

```diff
@@ -0,0 +1,102 @@
+    public required = false;
+
+    private _exporter: _Exporter;
+
+    private _wasUsed = false;
+
+    constructor(exporter: _Exporter) {
+        this._exporter = exporter;
+    }
+
+    public dispose() {
+    }
+
+    /** @hidden */
+    public get wasUsed() {
+        return this._wasUsed;
+    }
+
+    public postExportMaterialAdditionalTextures?(context: string, node: IMaterial, babylonMaterial: Material): BaseTexture[] {
+        let additionalTextures: BaseTexture[] = [];
+        if (babylonMaterial instanceof PBRMaterial) {
+            if (babylonMaterial.clearCoat.isEnabled) {
+                if (babylonMaterial.clearCoat.texture) {
+                    additionalTextures.push(babylonMaterial.clearCoat.texture);
+                }
+                if (!babylonMaterial.clearCoat.useRoughnessFromMainTexture && babylonMaterial.clearCoat.textureRoughness) {
+                    additionalTextures.push(babylonMaterial.clearCoat.textureRoughness);
+                }
+                if (babylonMaterial.clearCoat.bumpTexture) {
+                    additionalTextures.push(babylonMaterial.clearCoat.bumpTexture);
+                }
+                return additionalTextures;
+            }
+        }
+
+        return [];
+    }
+
+    public postExportMaterialAsync?(context: string, node: IMaterial, babylonMaterial: Material): Promise<IMaterial> {
+        return new Promise((resolve, reject) => {
+            if (babylonMaterial instanceof PBRBaseMaterial) {
+                if (!babylonMaterial.clearCoat.isEnabled) {
+                    resolve(node);
+                    return;
+                }
+
+                this._wasUsed = true;
+
+                if (node.extensions == null) {
+                    node.extensions = {};
+                }
+
+                let clearCoatTextureInfo = this._exporter._glTFMaterialExporter._getTextureInfo(babylonMaterial.clearCoat.texture);
+                let clearCoatTextureRoughnessInfo;
+                if (babylonMaterial.clearCoat.useRoughnessFromMainTexture) {
+                    clearCoatTextureRoughnessInfo = this._exporter._glTFMaterialExporter._getTextureInfo((babylonMaterial as PBRMaterial).metallicTexture
+                                                                                                        ?? (babylonMaterial as PBRMetallicRoughnessMaterial).metallicRoughnessTexture);
+                } else {
+                    clearCoatTextureRoughnessInfo = this._exporter._glTFMaterialExporter._getTextureInfo(babylonMaterial.clearCoat.textureRoughness);
+                }
+
+                let clearCoatNormalTextureInfo = this._exporter._glTFMaterialExporter._getTextureInfo(babylonMaterial.clearCoat.bumpTexture);
+
+                const clearCoatInfo: IKHRMaterialsClearcoat = {
+                    clearcoatFactor: babylonMaterial.clearCoat.intensity,
+                    clearcoatTexture: clearCoatTextureInfo ?? undefined,

```

Comment:
I don't think this is what you want.

`foo.bar = undefined` is not the same as `foo.bar` doesn't exist. I believe this will serialize into `{ bar: undefined }`.

Reply:
So far that doesn't seem to be the case, undefined/nulled fields appear to not be serialized. I'm not sure if thats a feature of the JSON serializer, or something we're intentionally doing.

Reply:
Yup, confirmed with JSON.stringify.